### ISSUE TITLE
Lint/RedundantSafeNavigation rule false positive for `send` method since 1.52.0

### ISSUE THREAD

**Issue by r-plus** [rubocop/rubocop#11927]
## Expected behavior

`Lint/RedundantSafeNavigation` rule will not offense for `send` method.

## Actual behavior

offense `nil&.send(:some_method)` code.

## Steps to reproduce the problem

test file

```ruby
# frozen_string_literal: true

nil&.send(:some_method) || 0
```

rubocop will offense this code.

```
$ rubocop ./test.rb
Inspecting 1 file
W

Offenses:

test.rb:3:4: W: [Correctable] Lint/RedundantSafeNavigation: Redundant safe navigation detected.
nil&.send(:some_method) || 0
   ^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## RuboCop version

```
$ rubocop -V
1.52.0 (using Parser 3.2.2.1, rubocop-ast 1.28.1, running on ruby 3.0.4) [arm64-darwin22]
```

## v1.51.0 is not offense

```
$ rubocop -V
1.51.0 (using Parser 3.2.2.1, rubocop-ast 1.28.1, running on ruby 3.0.4) [arm64-darwin22]

$ rubocop ./test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by koic** [comment]
This issue is a dup and resolved by #11915.

**Comment by r-plus** [comment]
oh, Thanks for letting me know.

**Comment by koic** [comment]
@bbatsov #11918 has the same issue and is already resolved in #11915. Can you cut the bug fix release at your convenience? (Or I'll take)
